mobilecoind-json/src/main.rs

```diff
@@ -0,0 +1,317 @@
+        .get_account_key(&req)
+        .map_err(|err| format!("Failed getting account key for entropy: {}", err))?;
+
+    let account_key = resp.take_account_key();
+
+    let mut req = mc_mobilecoind_api::AddMonitorRequest::new();
+    req.set_account_key(account_key);
+    req.set_first_subaddress(monitor.start);
+    req.set_num_subaddresses(monitor.number);
+    req.set_first_block(0);
+
+    let monitor_response = state
+        .mobilecoind_api_client
+        .add_monitor(&req)
+        .map_err(|err| format!("Failed adding monitor: {}", err))?;
+
+    Ok(Json(JsonMonitorResponse {
+        monitor_id: hex::encode(monitor_response.monitor_id),
+    }))
+}
+
+#[derive(Serialize, Default)]
+struct JsonBalanceResponse {
+    balance: u64,
+}
+
+/// Balance check using a created monitor and subaddress index
+#[get("/monitors/<monitor_hex>/balance/<index>")]
+fn balance(
+    state: rocket::State<State>,
+    monitor_hex: String,
+    index: u64,
+) -> Result<Json<JsonBalanceResponse>, String> {
+    let monitor_id =
+        hex::decode(monitor_hex).map_err(|err| format!("Failed to decode monitor hex: {}", err))?;
+
+    let mut req = mc_mobilecoind_api::GetBalanceRequest::new();
+    req.set_monitor_id(monitor_id.clone());
+    req.set_subaddress_index(index);
+
+    let resp = state
+        .mobilecoind_api_client
+        .get_balance(&req)
+        .map_err(|err| format!("Failed getting balance: {}", err))?;
+    let balance = resp.get_balance();
+    Ok(Json(JsonBalanceResponse { balance }))
+}
+
+#[derive(Serialize, Default)]
+struct JsonRequestResponse {
+    request_code: String,
+}
+
+/// Generates a request code without a balance or memo
+/// TODO: also add a POST that includes balance and memo 
+#[get("/monitors/<monitor_hex>/request-code/<index>")]
+fn request_code(
+    state: rocket::State<State>,
+    monitor_hex: String,
+    index: u64,
+) -> Result<Json<JsonRequestResponse>, String> {
+    let monitor_id =
+        hex::decode(monitor_hex).map_err(|err| format!("Failed to decode monitor hex: {}", err))?;
+
+    // Get our public address.
+    let mut req = mc_mobilecoind_api::GetPublicAddressRequest::new();
+    req.set_monitor_id(monitor_id.clone());
+    req.set_subaddress_index(index);
+
+    let resp = state
+        .mobilecoind_api_client
+        .get_public_address(&req)
+        .map_err(|err| format!("Failed getting public address: {}", err))?;
+
+    let public_address = resp.get_public_address().clone();
+
+    // Generate b58 code
+    let mut req = mc_mobilecoind_api::GetRequestCodeRequest::new();
+    req.set_receiver(public_address);
+    let resp = state
+        .mobilecoind_api_client
+        .get_request_code(&req)
+        .map_err(|err| format!("Failed getting request code: {}", err))?;
+
+    Ok(Json(JsonRequestResponse {
+        request_code: String::from(resp.get_b58_code()),
+    }))
+}
+
+#[derive(Deserialize)]
+struct JsonTransferRequest {
+    request_code: String,
+    amount: u64,
```

Comment:
amount_override? the request code could already contain an amount

Reply:
I'm a little uneasy about this, I added an additional function so people could see the requested amount but I think just blindly paying a request without specifying an amount feels dangerous.

Reply:
Agreed, that is suboptimal. However, in my opinion, this feels like a hack. I think the correct solution is to make this resemble the grpc API a bit more: add a read-request-code POST endpoint that returns {address, amount, memo} and have the transfer request receive address and amount.

Reply:
Ok, that is done now. You can pass the result of read-request-code to transfer, which also makes it more 1:1 with the gRPC API. The README reflects this.